### ISSUE TITLE
web3.eth.subscribe not working

### ISSUE THREAD

**Issue by johnnyboo** [ethereum/web3.js#1093]
Running geth with websocket:

    geth --fast --cache=512 --ws --wsorigins="*" --wsapi "db,eth,net,ssh,miner,web3,personal,admin"

Then connecting with web3 in a Node.js app:

    const Web3 = require('web3');

    var web3 = new Web3(new Web3.providers.WebsocketProvider("ws://localhost:8546"));

    console.log(web3.version);

    web3.eth.subscribe('pendingTransactions', function(err, res) {
	    console.log('Here')
	    console.log(err)
	    console.log(res)
    }).on('data', function(transaction) {
	    console.log('Here 2')
	    console.log(transaction)
    });

"Here", "Here 2" do not print to the console. The subscription does not seem to activate, nothing is printed after the version prints to the console. `web3.version` and other web3 commands work fine.

---

Geth version: 1.7.1-stable

Web3 version: 1.0.0-beta.22

**Comment by er3es** [comment]
I have the same issue. I don't get any messages for either pendingTransactions or newBlockHeaders subscriptions.

Geth version: 1.7.2-stable
Web3 version: 1.0.0-beta.23

**Comment by Lytigas** [comment]
I don't receive any callbacks on any subscriptions either.
Web3 version: 1.0.0-beta.24
TestRPC 4.1.3

**Comment by er3es** [comment]
@Lytigas @johnnyboo can you please try something for me? Connect using `wscat` to the socket and try these [commands](https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB). Are you getting any notifications back? If not it might be an issue with geth and not web3.

**Comment by troystribling** [comment]
I have seen the same thing with parity v1.8.0-beta.

**Comment by karalabe** [comment]
Please make sure you are synced up. While geth is doing its fast sync, the chain is incomplete, so there aren't any events being fired, since there's no existing events yet. Only when the chain sync fully will events start appearing.

**Comment by raahil190** [comment]
I have the same issue - it dosn't seem to work on geth 1.7.2 (fast sync) - has anybody been able to resolve this yet ?

**Comment by LiorRabin** [comment]
same issue here - 
geth version 1.7.2 (without fast sync)
web3 version 1.0.0-beta.24

`wscat -c ws://some.host.com:8546 `
returns:
> error: Error: unexpected server response (403)

**Comment by MrTibbles** [comment]
Have you ensured that you are passing `--wsorigins`, with the host you wish to use? As that 403 may be a failed handshake.

**Comment by LiorRabin** [comment]
@MrTibbles this is the command I'm running geth:
`geth --rpc --ws --wsaddr 0.0.0.0 --wsorigins="*" --testnet --rpcaddr 0.0.0.0`

**Comment by LiorRabin** [comment]
ok, so I've solved the issue running:

`geth --rpc --rpccorsdomain="*" --ws --wsorigins="*" --wsaddr 0.0.0.0 --testnet --rpcaddr 0.0.0.0`

Note: need to set BOTH rpccorsdomain & wsorigins

**Comment by MrTibbles** [comment]
Sorry, completely missed your initial response. I was close though, that being handshake failed :)

**Comment by shamoons** [comment]
Not working for me:


```
  const Web3 = require('web3');
  const web3 = new Web3(new Web3.providers.WebsocketProvider(process.env.GETH_WS_URL));
  console.log(process.env.GETH_WS_URL);
  const syncSubscription = web3.eth.subscribe('syncing')
    .on('data', (tx) => {
      console.log('tx', tx);
    });

  console.log(syncSubscription);
```

I see `here` and the `syncSubscription` object, but nothing in the `on('data`

**Comment by nivida** [comment]
This should be fixed with the PR #2000
